src/test/ui/borrowck/borrowck-for-loop-head-linkage.nll.stderr

```diff
@@ -5,7 +5,7 @@ LL |     for &x in &vector {
    |               -------
    |               |
    |               immutable borrow occurs here
-   |               immutable borrow used here, in later iteration of loop
```

Comment:
@nikomatsakis this change is for the worse, I don't remember you saying that with this approach some things were going to be worse but I couldn't find the discussion about it and if this is one of those cases or it's just an error in the code :).